modules/distribution-zones/src/integrationTest/java/org/apache/ignite/internal/rebalance/ItRebalanceTest.java

```diff
@@ -146,6 +158,48 @@ void assignmentsChangingOnNodeLeaveNodeJoin() throws Exception {
         );
     }
 
+    @Test
+    @WithSystemProperty(key = THREAD_ASSERTIONS_ENABLED, value = "false")
+    public void compactionAfterAbortedRebalance() throws InterruptedException {
+        createZone(ZONE_NAME, 1, 1);
+        createTestTable(TABLE_NAME, ZONE_NAME);
+
+        IgniteImpl igniteImpl = igniteImpl(0);
+
+        TableViewInternal table = unwrapTableViewInternal(igniteImpl.tables().table(TABLE_NAME));
+
+        PersistentPageMemoryTableStorage tableStorage = (PersistentPageMemoryTableStorage) table.internalTable().storage();
+
+        table.keyValueView().put(
+                null,
+                Tuple.create().set("id", 1),
+                Tuple.create().set("val", "value1")
+        );
+
+        CompletableFuture<Void> abortRebalance = tableStorage.startRebalancePartition(0)
+                        .thenCompose(v -> tableStorage.abortRebalancePartition(0));
+
+        assertThat(abortRebalance, willCompleteSuccessfully());
+
+        assertEquals(0, tableStorage.getMvPartition(0).lastAppliedIndex());
+
+        waitForCompaction(tableStorage);
+    }
+
+    private static void waitForCompaction(PersistentPageMemoryTableStorage tableStorage) throws InterruptedException {
+        PersistentPageMemoryDataRegion dataRegion = tableStorage.dataRegion();
+        CheckpointManager checkpointManager = dataRegion.checkpointManager();
+
+        CheckpointProgress checkpointProgress = checkpointManager.forceCheckpoint("");
+        assertThat(checkpointProgress.futureFor(CheckpointState.FINISHED), willCompleteSuccessfully());
+
+        checkpointManager.triggerCompaction();
```

Comment:
Why, if according to the algorithm after completing the checkpoint we will launch compaction?